### PR TITLE
[generator] Show proper errors when failing to compile something.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Also move the Driver.RunCommand method in mmp/mtouch to a separate file so
that the generator can re-use it.

### CODE REVIEW

File: docs/website/generator-errors.md [hunk 1]
```diff
@@ -25,6 +25,10 @@ The easiest way to get exact version information is to use the **Xamarin Studio*
 
 ### <a name='BI0001'/>BI0001: The .NET runtime could not load the {mi.ReturnType.Name} type. Message: {ex.Message}
 
+### <a name='BI0002'/>BI0002: Could not compile the API bindings.
+
+This error is shown when the generated failed to compile the API bindings.
```
Comment (spouliot) [inline docs/website/generator-errors.md]: `generated failed` ? missing `code` ?

Comment (spouliot) [reply docs/website/generator-errors.md]: or s/generated/generator ?

File: docs/website/generator-errors.md [hunk 2]
```diff
@@ -50,6 +54,10 @@ This usually indicates a bug in the binding generator; please file a new issue o
 <!-- 1xxx: code generation -->
 <!-- 10xx: errors -->
 
+### <a name='BI1000/>'BI1000: Could not compile the generated API bindings.
+
+This indicates a bug in Xamarin.iOS/Xamarin.Mac; please [submit an issue](https://github.com/xamarin/xamarin-macios/wiki/Submitting-Bugs-&-Suggestions).
```
Comment (spouliot) [inline docs/website/generator-errors.md]: ` and, if possible, attach the files that the C# compiler failed to compile` ?

Comment (rolfbjarne) [reply docs/website/generator-errors.md]: The generator automatically deletes those files (unless --tmpdir is passed to bgen), so asking customers for those files is probably going to cause more problems than it's worth.

File: docs/website/generator-errors.md (hunk 1 above)
Comment (rolfbjarne) [reply docs/website/generator-errors.md]: The second one is right :) fixed now.

**Pull request merged**